問題作成 number_of_increasing_sequences_between_two_sequences (#1185)
yosupo06/library-checker-problems: math/number_of_increasing_sequences_between_two_sequences/task.md

## @{keyword.statement}

@{lang.en}
Given integers $N,M$ and two non-negative integer sequences $A = \left( A_0, A_1, \cdots, A_{N-1} \right), B = \left( B_0, B_1, \cdots, B_{N-1} \right)$.

> Find the number of a non-negative integer sequences $x = \left( x_0, x_1, \cdots, x_{N-1} \right)$ which satisfies the following conditions, and print it $\bmod \ 998244353$.

Reviewer: ```suggestion
Find the number of a non-negative integer sequences $x = \left( x_0, x_1, \cdots, x_{N-1} \right)$ which satisfies the following conditions, and print it $\bmod \ @{param.MOD}$.
```

こうしておくと info.toml 内の定数が置換されます

- $0 \leq x_0 \leq \cdots \leq x_{N-1} < M$
- $A_i \leq x_i < B_i$

@{lang.ja}

> 整数 $N,M$ と長さ $N$ の非負整数列 $A = \left( A_0, A_1, \cdots, A_{N-1} \right), B = \left( B_0, B_1, \cdots, B_{N-1} \right)$ が与えられます。非負整数列 $x = \left( x_0, x_1, \cdots, x_{N-1} \right)$ で以下の条件を満たすものの個数を $\bmod \ 998244353$ で求めてください。

Reviewer: ```suggestion
整数 $N,M$ と長さ $N$ の非負整数列 $A = \left( A_0, A_1, \cdots, A_{N-1} \right), B = \left( B_0, B_1, \cdots, B_{N-1} \right)$ が与えられます。非負整数列 $x = \left( x_0, x_1, \cdots, x_{N-1} \right)$ で以下の条件を満たすものの個数を $\bmod \ @{param.MOD}$ で求めてください。
```

- $0 \leq x_0 \leq \cdots \leq x_{N-1} < M$
- $A_i \leq x_i < B_i$
@{lang.end}

## @{keyword.constraints}

> - $1 \leq N \leq 2^{17}$
> - $1 \leq M \leq 2^{17}$

Reviewer: ```suggestion
- $1 \leq N \leq @{param.N_M_MAX}$
- $1 \leq M \leq @{param.N_M_MAX}$
```
